### PR TITLE
Updates & revert functionality

### PR DESCRIPTION
**Pull request by DriftingSands**
**Requires `templateInfo.json` to root of both template & project**

Settings tab of site-details page checks templateInfo of project & template to see if a higher version is avalible.

User can choose to update, extra text in the dialog will inform the user if it is a major update that can potentially break block authoring.

Users can revert to last commit before update in the revert dialog. This undoes changes made in the blocks, icons & theme editors.

Test URLs:

Before: https://main--eds-self-service-website--headwire-edge-delivery.hlx.live/
After: https://updates--eds-self-service-website--headwire-edge-delivery.hlx.live/

### CODE REVIEW

File: blocks/site-details/site-details.js [hunk 1]
```diff
@@ -599,6 +599,151 @@ const createDocsEl = (html) => `
...
+          const revertUpdateResponse = await fetch(`${endpoint}revert/${project.projectSlug}`, {
+            method: 'POST',
+            headers: { ...headers, 'Content-Type': 'application/json' },
+            body: JSON.stringify({ sha: formData.get('update') }),
+          });
+          if (revertUpdateResponse.ok) {
+            revertUpdateDialog.renderDialog('<h3 class="centered-info">Project reverted successfully!</h3>');
+            // replace update button/up-to-date message, check for updates after 6sec
+            updateInfoDiv.innerHTML = '<h3>Project reverted successfully!</h3>';
+            setTimeout(() => rerenderUpdatesSection(updateInfoDiv, { project, headers }), 15000);
```
Comment (icaraps) [inline blocks/site-details/site-details.js:731]: Mhh looks suspicious :) 15s ??

File: blocks/site-details/site-details.js [hunk 2]
```diff
@@ -599,6 +599,151 @@ const createDocsEl = (html) => `
...
+          } else {
+            revertUpdateDialog.renderDialog(OOPS);
+          }
+          revertUpdateDialog.setLoading(false);
+        }
+      };
+    } else if (updateList.length === 0) {
+      dialogContent.innerHTML = '<h3>Revert Project</h3><h4>No updates to revert to.</h4>';
```
Comment (icaraps) [inline blocks/site-details/site-details.js:739]: Can you center it maybe ?

File: blocks/site-details/site-details.js [hunk 3]
```diff
@@ -599,6 +599,151 @@ const createDocsEl = (html) => `
...
+    } else {
+      dialogContent.innerHTML = '<h3>Revert Project</h3><h4>Could not get update information.</h4>';
```
Comment (icaraps) [inline blocks/site-details/site-details.js:741]: Can you center it maybe ?

File: styles/styles.css [hunk 4]
```diff
@@ -110,6 +110,11 @@ p {
   line-height: 22px;
 }
 
+p.warning,
```
Comment (icaraps) [inline styles/styles.css:113]: I noticed there's a habit of adding CSS into styles.css although it's not used across the site. So I'd say unless it's used in multiple places, don't put that CSS in here. 
On the other hand, we also have duplicated CSS across blocks which would make sense to move into styles.css.

**Comment by icaraps** [comment]
Did you add `templateInfo.json` to all templates already ?

File: blocks/site-details/site-details.js (hunk 1 above)
Comment (DriftingSands) [reply blocks/site-details/site-details.js:731]: I was getting the template version from the url in EDS. This takes some time to update, but I'll change it to get the contents from github. It just saved us some API calls, but I guess it's better to respond quickly

Comment (DriftingSands) [reply blocks/site-details/site-details.js]: This has been changed, along with the backend to always have the most accurate templateInfo be sent

File: styles/styles.css (hunk 4 above)
Comment (DriftingSands) [reply styles/styles.css]: .warning is now part of dialog styles

**Pull request merged**